library/L2_HAL/sensors/distance/timeofflight/tfmini.hpp

```diff
@@ -0,0 +1,227 @@
+  ///
+  /// @returns Status::kDeviceNotFound if device is not recognized
+  /// @returns Status::kBusError if data read from device is inconsistent
+  /// @returns Status::kSuccess if device is successfully read from
+  Status GetSignalStrengthPercent(float * strength) const override
+  {
+    Status success = sjsu::Status::kSuccess;
+    uint8_t device_data[kDeviceDataLength];
+
+    uart_pin_.Write(kPromptMeasurementCommand, kCommandLength);
+    uart_pin_.Read(device_data, kDeviceDataLength, kTimeout);
+    if ((device_data[0] != kFrameHeader) || (device_data[1] != kFrameHeader))
+    {
+      success   = sjsu::Status::kDeviceNotFound;
+      *strength = -1;
+    }
+    else
+    {
+      uint8_t checksum = 0;
+      for (int i = 0; i < 8; i++)
+      {
+        checksum += device_data[i];
+      }
+      if (checksum == device_data[8])
+      {
+        uint32_t stren = device_data[4];
+        stren |= device_data[5] << 8;
+        *strength = (stren / kStrengthUpperBound);
+      }
+      else
+      {
+        success   = sjsu::Status::kBusError;
+        *strength = -1;
+      }
+    }
+    return success;
+  }
+  /// Update the signal strength's lower limit valid range.
+  ///
+  /// @param lower_threshold - the value to set the lower threshold as.
+  ///        default value is 20, caps at 80. Decrease the value to increase the
+  ///        measurement range, increase value to improve reliability.
+  ///
+  /// @returns Status::kBusError if updating the TFMini fails
+  /// @returns Status::kSuccess if device is successfully updated
+  Status SetMinSignalThreshhold(uint8_t lower_threshold = 20)
+  {
+    Status device_updated = sjsu::Status::kSuccess;
+    uint8_t updated_min_threshold_command[kCommandLength];
+    uint8_t low_limit = lower_threshold;
+    if (low_limit > kStrengthLowerLimitCap)
+    {
+      low_limit = kStrengthLowerLimitCap;
+    }
+    for (int i = 0; i < kCommandLength; i++)
+    {
+      updated_min_threshold_command[i] = kUpdateMinThresholdCommand[i];
+    }
+    updated_min_threshold_command[kThresholdByte] = low_limit;
+    if (!SendCommandAndCheckEcho(kConfigCommand))
+      device_updated = sjsu::Status::kBusError;
+    if (!SendCommandAndCheckEcho(updated_min_threshold_command))
+      device_updated = sjsu::Status::kBusError;
+    if (!SendCommandAndCheckEcho(kExitConfigCommand))
+      device_updated = sjsu::Status::kBusError;
+
+    if (device_updated == sjsu::Status::kSuccess)
+      min_threshold_ = lower_threshold;
+
+    return device_updated;
+  }
+
+ private:
+  const Uart & uart_pin_;
+  uint8_t min_threshold_ = 20;
+  bool SendCommandAndCheckEcho(const uint8_t * command) const
+  {
+    bool success = true;
+    uint8_t echo[kCommandLength];
+
+    uart_pin_.Write(command, kCommandLength);
+    uart_pin_.Read(echo, kCommandLength, kTimeout);
+    for (int i = 0; i < kCommandLength; i++)
+    {
+      if ((kEchoSuccess[i] != echo[i]) && (kConfigExit[i] != echo[i]))
+        success = false;
```

Comment:
Use curly braces for all conditional blocks. Singular blocks are not acceptable in SJSU-Dev2.